omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs2_0;
+
+import org.openmrs.Allergy;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestTestConstants2_0;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Ignore;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.WebRequest;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+public class PatientAllergyController2_0Test extends MainResourceControllerTest {
+	
+	@Before
+	public void init() throws Exception {
+		executeDataSet(RestTestConstants2_0.ALLERGY_TEST_DATA_XML);
+	}
+	
+	@Override
+	public String getURI() {
+		return "patient/" + RestTestConstants2_0.PATIENT_UUID + "/allergy";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_ID;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 0;
+	}
+	
+	@Test
+	public void deleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergy(1);
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + RestTestConstants2_0.ALLERGY_ID, new Parameter("reason", "unit test")));
+		allergy = Context.getPatientService().getAllergy(1);
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	@Test
+	@Ignore("REST API does not support delete /patient/patient_uuid/allergy/ to delete all allergies")
+	public void deleteAllAllergies() throws Exception
+	{
+		handle(newDeleteRequest(getURI() + "/", new Parameter("reason", "unit test")));
```

Comment:
PS- @jdegraft if this particular thing is especially hard, then feel free to complete everything else, but leave this undone and create another ticket for it.